pkgs/development/python-modules/sure/default.nix

```diff
@@ -23,17 +23,20 @@ buildPythonPackage rec {
 
   postPatch = ''
     substituteInPlace setup.cfg \
-      --replace "rednose = 1" ""
+      --replace "rednose = 1" "" \
+      --replace-fail "--cov=sure" ""
   '';
 
-  propagatedBuildInputs = [
+  build-system = [ setuptools ];
+  dependencies = [ six ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
     mock
-    six
   ];
-
-  doCheck = pythonOlder "3.12"; # nose requires imp module
-
-  nativeCheckInputs = [ nose ];
+  preCheck = ''
+    rm tests/test_old_api.py # require nose
+  '';
```

Comment:
Can you use `disabledTests`/`disabledTestPaths` here? (Nit: Please add a newline before the definition, as well.)